service/src/main/java/skills/skillLoading/SkillsLoader.groovy

```diff
@@ -1045,7 +1046,7 @@ class SkillsLoader {
                         maxOccurrencesWithinIncrementInterval: skillDef.numMaxOccurrencesIncrementInterval,
                         totalPoints: skillDef.totalPoints,
                         dependencyInfo: skillDefAndUserPoints.dependencyInfo,
-                        selfReporting: skillDef.selfReportingType ? new SelfReportingInfo(enabled: true, type: skillDef.selfReportingType, justificationRequired: Boolean.valueOf(skillDef.justificationRequired)) : null,
+                        selfReporting: skillDef.selfReportingType ? loadSelfReporting(userId, skillDef) : null,
```

Comment:
- ``loadSelfReporting`` will get called for each skill under a subject which internally calls out to DB for data - would cause long latency for subjects with many skills; the required data should be passed in to ``createSkillSummaries`` method via  ``childrenWithPoints`` param; can probably load all of the self-report requests under a subject in the ``skills.skillLoading.SubjectDataLoader#loadData`` method and then associate it to ``skillsAndPoints`` result
- add service endpoints tests when loading subject, badges and global badges to verify that approval requests are properly returned
- cypress test to verify that ``Pending Approval`` is rendered or not rendered